src/prisma/prisma.service.ts

```diff
@@ -3,7 +3,30 @@ import { PrismaClient } from '@prisma/client';
 
 @Injectable()
 export class PrismaService extends PrismaClient implements OnModuleInit {
+  private readonly maxRetries = 5;
+  private readonly retryDelay = 2000;
   async onModuleInit() {
-    await this.$connect();
+    await this.retryConnect();
+  }
+
+  private async retryConnect(attempt = 1): Promise<void> {
```

Comment:
**praise**: Implementing a retry mechanism for database connection is a robust solution to handle transient connection issues.